src/mpyl/utilities/repo/__init__.py

```diff
@@ -244,52 +249,58 @@ def changes_between(self, base_revision: str, head_revision: str) -> list[Revisi
         changed_files = self._repo.git.diff(
             f"{base_revision}..{head_revision}", name_only=True
         )
-        return Revision.from_git_output(revs, changed_files)
+        return Changeset.from_git_output(revs, changed_files)
 
-    def changes_in_branch(self) -> list[Revision]:
-        base_ref = self.base_revision
-        base_hex = base_ref.hexsha if base_ref else self.root_commit_hex
+    def changes_in_branch(self) -> Changeset:
+        return Changeset(self.get_sha, self.changed_files_in_branch())
 
-        head_hex = self._repo.active_branch.commit.hexsha
-
-        return self.changes_between(base_hex, head_hex)
+    def changed_files_in_branch(self) -> set[str]:
+        # TODO pass the base_branch as a build parameter, not all branches  # pylint: disable=fixme
```

Comment:
TODO for when? Can we add the ticket number here?

Reply:
We don't have a ticket, it's planned for 1.6.1 - Pedro has a list of things we still need to do. Maybe discuss on Slack if we need tickets

Reply:
We should have tickets, I haven't had the time to create them yet.